File: frontend/src/temp.py
import pickle
import os.path
from google.auth.transport.requests import Request
from google_auth_oauthlib.flow import InstalledAppFlow
from googleapiclient.discovery import build

# Define the necessary scopes
SCOPES = ["https://www.googleapis.com/auth/calendar.events"]

# Load or create credentials
def get_credentials():
    creds = None
    # Check if token.pickle exists (stores user's access and refresh tokens)
    if os.path.exists("token.pickle"):
        with open("token.pickle", "rb") as token:
            creds = pickle.load(token)

    # Refresh or request new credentials
    if not creds or not creds.valid:
        if creds and creds.expired and creds.refresh_token:
            creds.refresh(Request())
        else:
            flow = InstalledAppFlow.from_client_secrets_file(
                "client_secret.json", SCOPES
            )
            creds = flow.run_local_server(port=5173)  # Ensure port matches redirect_uri

        # Save the credentials for future use
        with open("token.pickle", "wb") as token:
            pickle.dump(creds, token)

    return creds

# Function to create Google Meet event
def create_google_meet():
    creds = get_credentials()
    service = build("calendar", "v3", credentials=creds)

    event = {
        "summary": "Google Meet Meeting",
        "start": {
            "dateTime": "2025-03-26T10:00:00",
            "timeZone": "Asia/Kolkata",
        },
        "end": {
            "dateTime": "2025-03-26T11:00:00",
            "timeZone": "Asia/Kolkata",
        },
        "conferenceData": {
            "createRequest": {
                "requestId": "meet-request-123",
                "conferenceSolutionKey": {"type": "hangoutsMeet"},
            }
        },
    }

    event_result = service.events().insert(
        calendarId="primary",
        body=event,
        conferenceDataVersion=1,
    ).execute()

    meet_link = event_result.get("hangoutLink")
    print(f"✅ Google Meet Link: {meet_link}")

# Run the function
create_google_meet()

File: backend/temp.js
import express from "express";
import { google } from "googleapis";
import dotenv from "dotenv";

dotenv.config({ path: "../.env" });
const app = express();
const PORT = 5173;

// 🔹 Load OAuth Credentials
const CLIENT_ID = process.env.CLIENT_ID;
const CLIENT_SECRET = process.env.CLIENT_SECRET;
const REDIRECT_URI = process.env.REDIRECT_URI;
const ACCESS_TOKEN = process.env.ACCESS_TOKEN;
const REFRESH_TOKEN = process.env.REFRESH_TOKEN;

console.log("CLIENT_ID", CLIENT_ID);
console.log("CLIENT_SECRET", CLIENT_SECRET);  
// 🔹 Initialize OAuth2 Client
const oauth2Client = new google.auth.OAuth2(CLIENT_ID, CLIENT_SECRET, REDIRECT_URI);
oauth2Client.setCredentials({
  access_token: ACCESS_TOKEN,
  refresh_token: REFRESH_TOKEN,
});

// 🔹 Create Google Meet Link
app.get("/create-meet", async (req, res) => {
  try {
    const calendar = google.calendar({ version: "v3", auth: oauth2Client });

    // Define event details
    const event = {
      summary: "Google Meet Meeting",
      description: "Discussing project updates",
      start: { dateTime: new Date().toISOString(), timeZone: "Asia/Kolkata" },
      end: { dateTime: new Date(Date.now() + 3600000).toISOString(), timeZone: "Asia/Kolkata" },
      conferenceData: {
        createRequest: { requestId: `meet-${Date.now()}` },
      },
    };

    // Insert event into Google Calendar
    const response = await calendar.events.insert({
      calendarId: "primary",
      resource: event,
      conferenceDataVersion: 1,
    });

    console.log("📅 Meet Link:", response.data.hangoutLink);
    res.json({ meetLink: response.data.hangoutLink });
  } catch (error) {
    console.error("❌ Error creating Meet link:", error);
    res.status(500).send("Failed to create Meet link.");
  }
});

// Start the server
app.listen(PORT, () => {
  console.log(`🚀 Server running on http://localhost:${PORT}`);
});
